jsonrpc/eth_endpoint.go

```diff
@@ -665,10 +665,10 @@ func (e *Eth) EstimateGas(arg *txnArgs, rawNum *BlockNumber) (interface{}, error
 				// Specifying the transaction failed, but not providing an error
 				// is an indication that a valid error occurred due to low gas,
 				// which will increase the lower bound for the search
-				return true, nil
+				return true, []byte(hex.EncodeToString(result.ReturnValue)), nil
```

Comment:
If I am correct, if the `applyErr` is not nil, it could happen that `result` is nil, so it is not safe to use it inside `if applyErr != nil {` branch without checking for nil first.